packages/rOpenScPCA/tests/testthat/test-sweep-clusters.R

```diff
@@ -0,0 +1,123 @@
+suppressPackageStartupMessages(library(SingleCellExperiment))
+
+set.seed(2024)
+sce <- splatter::simpleSimulate(nGenes = 1000, verbose = FALSE) |>
+  scater::logNormCounts() |>
+  scater::runPCA(ncomponents = 10)
+
+
+test_that("sweep_clusters works as expected with defaults", {
+  sweep_list <- sweep_clusters(
+    sce,
+    nn = c(10, 15),
+    resolution = c(0.5, 1)
+  )
+
+  expect_equal(
+    length(sweep_list),
+    4
+  )
+
+  # check colnames one at a time
+  sweep_list |>
+    purrr::walk(
+      \(df) {
```

Comment:
Rather than nesting this, we could bind rows and then do the tests? I don't think it matters much.

Reply:
I actually had that originally for all tests _except_ column names since we definitely want to be sure about those on an individual-df level. I ended up moving everything into the `map` since I thought that was a smidge, but really only a smidge..., more precise